src/evotorch/algorithms/functional/funcdcem.py

```
@@ -0,0 +1,470 @@
+    else:
+        if num_choices is None:
+            raise ValueError(
+                "When the population consists of non-boolean integer variables, `num_choices` must be at least 2."
+                " However, it is omitted (i.e. left as None)."
+            )
+        num_choices = int(num_choices)
+        if num_choices < 2:
+            raise ValueError(
+                "When the population consists of non-boolean integer variables, `num_choices` must be at least 2."
+                " However, it is omitted (i.e. left as None)."
+            )
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)